ci_framework/plugins/action/ci_kustomize.py

```diff
@@ -0,0 +1,731 @@
+    - If I(target_path) points to a file 'cifmw-kustomization-result.yaml'
+      in I(target_path) will be used.
+    type: str
+  extra_dirs:
+    description:
+    - Additional paths where Kustomizations should be searched.
+    type: list
+    elements: str
+  preserve_workspace:
+    description:
+    - If true, the workspace is not deleted if success.
+    - If failure this option is ignored and the workspace is preserved.
+    type: bool
+    default: false
+  kustomization_files_goes_first:
+    description:
+    - If true, Kustomizations given by I(kustomizations) are applied before
+      the ones from filesystem.
+    type: bool
+    default: true
+  sort_ascending:
+    description:
+    - If true, file Kustomizations are ordered by ascending order of the file
+      name. Descnding orther oherwise.
+      the ones from filesystem.
+    type: bool
+    default: true
+"""
+
+EXAMPLES = r"""
+# Apply the kustomizations in `/home/user/source/k8s-manifets-dir` to the
+# `target_path` manifest and output the result in `output_pat`
+- name: Apply the file and variables kustomizations to multiple CRs
+  ci_kustomize:
+    target_path: /home/user/source/k8s-manifets-dir/manifest.yaml
+    output_path: /home/user/source/k8s-manifets-dir/out.yaml
+
+# Apply the given kustomizations in the kustomizations variable and in
+# `/home/user/source/k8s-manifets-dir` and `extra_dir` dirs to the
+# manifests available in the `target_path` dir
+- name: Apply the file and variables kustomizations to multiple CRs
+  ci_kustomize:
+    target_path: /home/user/source/k8s-manifets-dir
+    kustomizations:
+      - apiVersion: kustomize.config.k8s.io/v1beta1
+        kind: Kustomization
+        patches:
+          - patch: |-
+              - op: replace
+                path: /metadata/labels/release
+                value: "1.2.3.4"
+            target:
+              kind: Deployment
+      - |-
+        ---
+        apiVersion: kustomize.config.k8s.io/v1beta1
+        kind: Kustomization
+        patches:
+        -   patch: |-
+              - op: add
+                path: /metadata/labels/app
+                value: "my-app"
+            target:
+              kind: Deployment
+        -   patch: |-
+              - op: add
+                path: /metadata/labels/app
+                value: "my-app"
+            target:
+              kind: ConfigMap
+        ---
+        apiVersion: kustomize.config.k8s.io/v1beta1
+        kind: Kustomization
+        patches:
+        -   patch: |-
+              - op: add
+                path: /metadata/annotations/imageregistry
+                value: "https://hub.docker.com/"
+            target:
+              kind: Deployment
+    extra_dirs:
+      - /home/user/source/prod-kustomizations
+"""
+
+RETURN = r"""
+count:
+    description: Total number of Kustomizations applied
+    returned: success
+    type: int
+    sample: 10
+kustomizations_paths: typing.List[str]
```

Comment:
I think it should be `kustomizations_paths:` without typing.List[str]

Reply:
Good catch!